### PR TITLE
Calculated cell: Fix zero value rendering issue

### LINKED ISSUE #2895: Enhancement- unknown should show as 0.00 - tab 4a
@minotogna 
When the **unknown**(d) is 0.00, it should show the value 0.00 in tab 4a instead of being empty.

Tab 1a:

Tab 4a:

### PR DESCRIPTION
**Pull request by sorja**
image

### CODE REVIEW

File: src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Calculated/Calculated.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { PropsCell } from '../props'
 const Calculated: React.FC<PropsCell> = (props) => {
   const { nodeValue, row } = props
 
-  const value = nodeValue?.raw ? Numbers.format(Number(nodeValue.raw), row.props?.format?.integer ? 0 : 2) : ''
+  const value = Numbers.format(nodeValue.raw) ? Numbers.format(nodeValue.raw, row.props?.format?.integer ? 0 : 2) : ''
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Calculated/Calculated.tsx]: This is strange . Why is this happening ?

Comment (sorja) [reply src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Calculated/Calculated.tsx]: I tested it:

### solution
Numbers.format(0) => 0.00

Numbers.format(null) => null

Numbers.format(undefined) => null

### issue
` const value = nodeValue?.raw ? Numbers.format(Number(nodeValue.raw), row.props?.format?.integer ? 0 : 2) : ''`

When calculation formula returns 0 ( eg. here: unknown = `forestArea - A - B` where `forestArea = 15` and `A = 5` and `B = 10` nodeValue.

( unknown = 0 = 15 - 10 - 5 )

therefore

` const value = nodeValue?.raw ( here 0 = false ) ? Numbers.format(Number(nodeValue.raw), row.props?.format?.integer ? 0 : 2) : ''`

evaluated to

`const value = ''`

even though we want

`const value = 0.00`

Comment (minotogna) [reply src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Calculated/Calculated.tsx]: ok, but the double format is strange to read. can u find a better way? Like Number.isNan ?

File: src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Calculated/Calculated.tsx [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 import { Numbers } from 'utils/numbers'
+import { Objects } from 'utils/objects'
 
 import { PropsCell } from '../props'
 
 const Calculated: React.FC<PropsCell> = (props) => {
   const { nodeValue, row } = props
 
-  const value = nodeValue?.raw ? Numbers.format(Number(nodeValue.raw), row.props?.format?.integer ? 0 : 2) : ''
+  const value = Objects.isEmpty(nodeValue.raw) ? Numbers.format(nodeValue.raw, row.props?.format?.integer ? 0 : 2) : ''
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Calculated/Calculated.tsx]: are you sure if it's empty you format it ?

Comment (sorja) [reply src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Calculated/Calculated.tsx]: Good catch, typo

Now it works as expected!

image

**Review by minotogna — DISMISSED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**